pkg/cmd/repo/view/http.go

```diff
@@ -0,0 +1,41 @@
+package view
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+)
+
+type RepoReadme struct {
+	Filename string
+	Content  string
+}
+
+func RepositoryReadme(client *api.Client, repo ghrepo.Interface) (*RepoReadme, error) {
+	var response struct {
+		Name    string
+		Content string
+	}
+
+	err := client.REST("GET", fmt.Sprintf("repos/%s/readme", ghrepo.FullName(repo)), nil, &response)
+	if err != nil {
+		var httpError api.HTTPError
+		if errors.As(err, &httpError) && httpError.StatusCode == 404 {
+			return nil, api.NewNotFoundError(err)
```

Comment:
I wonder whether we need to use `api.NotFound` at all here! The only consumer that will check for this specific "not found" case is the `viewRun()` function. Could we get away with returning a static "not found" object specific to this package?

```go
var NotFoundError = errors.New("not found")
```